### PR TITLE
fix: Ensure ParquetTools.readTable attaches the inferred definition

### PR DESCRIPTION
**Pull request by devinrsmith**
Fixes an inconsistency between `ParquetTools#readSingleFileTable` and `ParquetTools#readTable` in regards to whether they set the TableDefinition in the ParquetInstructions after inference.

This PR is the non-controversial part extracted from #6149

### CODE REVIEW

File: extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetInstructions.java [hunk 1]
```diff
@@ -35,6 +35,19 @@ public abstract class ParquetInstructions implements ColumnToCodecMappings {
...
+     * Throws an exception if {@link ParquetInstructions#getTableDefinition()} is empty.
+     *
+     * @param parquetInstructions the parquet instructions
+     * @throws IllegalArgumentException if there is not a table definition
+     */
+    public static TableDefinition ensureDefinition(ParquetInstructions parquetInstructions) {
+        if (parquetInstructions.getTableDefinition().isEmpty()) {
+            throw new IllegalArgumentException("Table definition must be provided");
+        }
+        return parquetInstructions.getTableDefinition().get();
```
Comment (rcaudy) [inline extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetInstructions.java:48]:
```suggestion
        return readInstructions.getTableDefinition().orElseThrow(
                () -> new IllegalArgumentException("Table definition must be provided"));
```

**Review by rcaudy — COMMENTED** [review]
Some minor comments.

File: extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetTableLocationFactory.java [hunk 2]
```diff
@@ -25,7 +26,8 @@ public final class ParquetTableLocationFactory implements TableLocationFactory<T
     private final ParquetInstructions readInstructions;
 
     public ParquetTableLocationFactory(@NotNull final ParquetInstructions readInstructions) {
-        this.readInstructions = readInstructions;
+        this.readInstructions = Objects.requireNonNull(readInstructions);
+        ParquetInstructions.ensureDefinition(readInstructions);
```
[the remark this answers is not in the document]
Comment (devinrsmith) [reply extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetTableLocationFactory.java:30]: Oops, meant to revert this part.

File: extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java [hunk 3]
```diff
@@ -857,11 +856,10 @@ public static Table readTable(
         if (readInstructions.getTableDefinition().isEmpty()) {
             // Infer the definition
             final KnownLocationKeyFinder<ParquetTableLocationKey> inferenceKeys = toKnownKeys(locationKeyFinder);
-            final Pair<TableDefinition, ParquetInstructions> inference = infer(inferenceKeys, readInstructions);
+            useInstructions = infer(inferenceKeys, readInstructions);
```
[the remark this answers is not in the document]
Comment (devinrsmith) [reply extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java:859]: I'm not sure what you mean.

**Review by rcaudy — APPROVED** [review]

**Pull request merged**